src/coreclr/gc/gc.cpp

```diff
@@ -11883,16 +11898,64 @@ void region_free_list::add_region_front (heap_segment* region)
     head_free_region = region;
     heap_segment_prev_free_region (region) = nullptr;
 
-    num_free_regions++;
-    num_free_regions_added++;
+    update_added_region_info (region);
+}
 
-    size_t region_size = get_region_size (region);
-    size_free_regions += region_size;
+// This goes forward in the list till we find a region whose committed size is <= this region's committed
+// or we reach the tail.
+void region_free_list::add_region_in_descending_order (heap_segment* region_to_add)
```

Comment:
this might require some optimization. I just wrote it in a primitive way. are you seeing this show up in profiles if you do a lot of returning to the region free list?

Reply:
No, it didn't show up in profiles. I was worried about it too, but I think in most cases, we'll return fully committed regions, and in that case, the insertion will be fast. But we should run profiles on test cases with more regions.

Reply:
Ran some profiles, found that it was inlined. Made it NOINLINE, found a small cost in it (10 samples out of 120,000). I changed it so that fully committed regions would be inserted at the front, and we would search **backwards** for not fully committed regions. The idea is that if there are a lot of regions on the list, most will be fully committed.